cli/cluster.go

```diff
@@ -195,3 +206,60 @@ func runClusterDefault(args *docopt.Args) error {
 	log.Printf("%q is now the default cluster.", name)
 	return nil
 }
+
+func runClusterMigrateDomain(args *docopt.Args) error {
+	client, err := getClusterClient()
+	if err != nil {
+		shutdown.Fatal(err)
+	}
+
+	dm := &ct.DomainMigration{
+		Domain: args.String["<domain>"],
+	}
+
+	release, err := client.GetAppRelease("controller")
+	if err != nil {
+		return err
+	}
+	dm.OldDomain = release.Env["DEFAULT_ROUTE_DOMAIN"]
+
+	if !promptYesNo(fmt.Sprintf(`Migrate cluster domain from "%s" to "%s"?`, dm.OldDomain, dm.Domain)) {
+		fmt.Println("Aborted")
+		return nil
+	}
+
+	events := make(chan *ct.Event)
+	stream, err := client.StreamEvents(controller.StreamEventsOptions{
+		ObjectTypes: []ct.EventType{ct.EventTypeDomainMigration},
+	}, events)
+	if err != nil {
+		return nil
+	}
+	defer stream.Close()
+
+	if err := client.PutDomain(dm); err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case event, ok := <-events:
+			if !ok {
+				return stream.Err()
+			}
+			var e *ct.DomainMigrationEvent
+			if err := json.Unmarshal(event.Data, &e); err != nil {
+				return err
+			}
+			if e.Error != "" {
+				fmt.Println(e.Error)
+			}
+			if e.DomainMigration.FinishedAt != nil {
+				fmt.Printf("Changed cluster domain from \"%s\" to \"%s\"\n", dm.OldDomain, dm.Domain)
+				return nil
+			}
+		case <-time.After(2 * time.Minute):
```

Comment:
Given this could take up to 2 mins without giving any output, we may want to warn the user so they don't `ctrl-c`, e.g.:

```
migrating domain (this can take up to 2 minutes)...
```

or we could print something every few seconds (e.g. `.`).